participant/app/utils/validation.ts

```diff
@@ -91,40 +91,52 @@ export const participantSchema = zfd.formData({
       ),
       preferredName: zfd.text(z.string().optional()),
 
-      dob: z.object({
-        day: zfd.numeric(
-          z
-            .number()
-            .min(1, {
-              message:
-                "Date of birth must include a valid month, day, and year.",
-            })
-            .max(31, {
-              message:
-                "Date of birth must include a valid month, day, and year.",
-            })
-        ),
-        month: zfd.numeric(
-          z
-            .number()
-            .min(1, {
+      dob: z
+        .object({
+          day: zfd.numeric(),
+          month: zfd.numeric(),
+          year: zfd.numeric(),
+        })
+        .superRefine((val, ctx) => {
+          if (!val.year || !val.month || !val.day) {
+            ctx.addIssue({
+              code: z.ZodIssueCode.custom,
               message:
                 "Date of birth must include a valid month, day, and year.",
-            })
-            .max(12, {
+            });
+            return z.NEVER;
+          }
+          const date = new Date(val.year, val.month - 1, val.day);
```

Comment:
Why `month - 1`?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getMonth
Month is zero indexed?

Reply:
💀